### ISSUE TITLE
Cursors context getting lost in Admin UI

### ISSUE THREAD

**Issue by hlcianfagna** [crate/crate#13482]
### CrateDB version

5.3.0 -SNAPSHOT-928d076

### CrateDB setup information

```bash
docker run --publish 4200:4200 --publish 5432:5432 crate/crate:nightly-5.3.0-20230117 -Cdiscovery.type=single-node
```
on Ubuntu Server 22.04.1 LTS
running on VirtualBox 7.0.4 with a bridged network adapter
ufw disabled

### Problem description

The "transaction" context necessary to work with cursors `WITHOUT HOLD` seems to be getting lost when using the Admin UI, there are no issues when using `crash`.

### Steps to Reproduce

Use a web browser from outside the VM which runs CrateDB, navigate to the Admin UI on port 4200
```sql
BEGIN;

DECLARE cursor1 NO SCROLL CURSOR FOR SELECT * FROM generate_series(1,10);

FETCH 10 FROM cursor1;

DECLARE cursor2 NO SCROLL CURSOR FOR SELECT * FROM generate_series(11,20);
```

### Actual Result

```sql
FETCH 10 FROM cursor1;
```
--> `io.crate.exceptions.SQLParseException: No cursor named `cursor1` available`

```sql
DECLARE cursor2 NO SCROLL CURSOR FOR SELECT * FROM generate_series(11,20);
```
--> `io.crate.exceptions.UnsupportedFeatureException: DECLARE CURSOR can only be used in transaction blocks`

### Expected Result

`FETCH 10 FROM cursor1;` pulls 10 rows and the declaration of `cursor2` succeeds like when using `crash` also from outside the VM serving CrateDB.

**Comment by mfussenegger** [comment]
The admin interface afaik doesn't keep the connection open. Each query results in a new session. 

Was there a concrete use case for using cursors in the admin interface or was this merely playing around/testing?

It probably makes sense to change the behaviour, but having more concrete use cases could boost priority

**Comment by hlcianfagna** [comment]
Yes, @proddata had explained that, he also mentioned that something similar can be observed with session settings.
I am not aware of any customers encountering this.
I was just logging it as we discussed quickly yesterday to track that this is something that would be unexpected from a user perspective, but I understand it would not be easy to address, just something to keep in mind if a refactor of the Admin UI is considered at some point.

**Comment by proddata** [comment]
> The admin interface afaik doesn't keep the connection open. Each query results in a new session.

That afaik is not entirely true. The behaviour is somewhat flaky however.

```
SET application_name TO 'test';
```

if you run
```
SHOW application_name;
```

this alters between `'test'` and `null`

**Comment by hlcianfagna** [comment]
Yes, it is not exactly one query one session at the moment. Notice that in the repro steps in this ticket the declaration of `cursor1` succeeded.
